### ISSUE TITLE
Headers on Android lose accessibility ability when headerTransparent: true

### ISSUE THREAD

**Issue by helblinglilly** [react-community/react-navigation#13043]
### Current behavior

When the header has the option `headerTransparent: true`, then the entire header becomes inaccessible to screen readers on Android.

TalkBack cannot navigate into the header anymore.

This does not exist on iOS - VoiceOver will read out and select the headers fine.

Screenshots picture use of `scrcpy` which highlights the select a11y element, but can be reproduced with native TalkBack options as well.

https://github.com/user-attachments/assets/5a6eeb64-300d-4e57-9ae9-c0b0c35f953d

### Expected behavior

The behaviour should be the same as iOS, or when `headerTransparent: false` whereby the header has the first tab index, can can be navigated to by swiping on the screen.

https://github.com/user-attachments/assets/3caedb62-4053-4244-8e68-e3dc9c71121d

### Reproduction

https://snack.expo.dev/x9MuliDzjthfDDLt1Hx1Q

### Platform

- [x] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [x] @react-navigation/stack
- [ ] @react-navigation/native-stack
- [ ] react-native-drawer-layout
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 7.1.34|
| @react-navigation/stack                | 7.8.6|
| react-native-safe-area-context         |~5.6.0|
| react-native-gesture-handler           |~2.28.0|
| expo                                   |54.0.0|

**Comment by helblinglilly** [comment]
I can't access those latest versions via the Snack yet, but can confirm that the issue can still be reproduced on those versions

**Comment by lcsy1234** [comment]
I want to try fixing this bug, and recently I'm planning to submit a pull request.

**Comment by matt-dalton** [comment]
Did this ever get resolved? We still seem to be getting something like this happening on our E2E tests
